frontend/src/components/ShootDns/GManageDns.vue

```diff
@@ -5,158 +5,314 @@ SPDX-License-Identifier: Apache-2.0
                 </template>
-                <v-list-item-subtitle>
-                  Type: {{ item.raw.type }}
-                </v-list-item-subtitle>
               </v-list-item>
             </template>
             <template #selection="{ item }">
-              <g-vendor-icon :icon="item.raw.type" />
-              <span class="ml-2">
-                {{ item.raw.secretName }}
-              </span>
+              <div class="d-flex">
+                <g-vendor-icon
+                  :icon="item.value"
+                  class="mr-2"
+                />
+                {{ item.value }}
+              </div>
             </template>
           </v-select>
         </v-col>
+        <v-col cols="3">
+          <g-select-secret
+            v-model="primaryDnsProviderSecret"
+            :dns-provider-kind="dnsPrimaryProviderType"
+            register-vuelidate-as="dnsProviderSecret"
+            label="Primary DNS Provider Secret"
+          />
+        </v-col>
       </v-row>
-    </v-expand-transition>
-    <v-row>
-      <v-divider class="mx-3" />
-    </v-row>
-  </template>
-  <div class="alternate-row-background">
-    <v-expand-transition group>
-      <v-row
-        v-for="id in dnsProviderIds"
-        :key="id"
-        class="list-item pt-2"
-      >
-        <g-dns-provider-row :dns-provider-id="id" />
-      </v-row>
-    </v-expand-transition>
-    <v-row
-      key="addProvider"
-      class="list-item my-1"
-    >
+    </template>
+    <v-row v-if="domainRecommendationVisible">
       <v-col>
-        <v-btn
-          variant="text"
+        <v-alert
+          variant="tonal"
+          type="info"
           color="primary"
-          @click="addDnsProvider"
+          max-width="50%"
         >
-          <v-icon class="text-primary">
-            mdi-plus
-          </v-icon>
-          Add DNS Provider
-        </v-btn>
+          <div>
+            The cluster domain is used for the API server of this cluster.
+            To expose workload services with your cluster domain, add a DNS provider for the <code>shoot-dns-service</code> extension.
+            Click the <span class="font-weight-bold">Apply</span> button below to configure a provider for the <code>shoot-dns-service</code> extension using your cluster domain as the included domain.
+          </div>
+          <v-btn
+            class="mt-3"
+            size="small"
+            color="primary"
+            @click="addDnsServiceExtensionProviderForCustomDomain"
+          >
+            Apply
+          </v-btn>
+        </v-alert>
       </v-col>
     </v-row>
-  </div>
+    <template v-if="hasDnsServiceExtension">
+      <div class="text-subtitle-1 my-3">
+        DNS Providers for the shoot-dns-service Extension
+      </div>
+      <div class="text-caption">
+        Configure DNS providers for the shoot-dns-service-extension to expose cluster workloads using domains managed by the configured providers
```

Comment:
```suggestion
        Configure DNS providers for the shoot-dns-service extension to automatically manage and synchronize DNS entries for cluster resources like services and ingresses
```
also, in the original code it was `shoot-dns-service-extension` instead of `shoot-dns-service extension`